qualang_tools/voltage_gates/README.md

```diff
@@ -0,0 +1,202 @@
+- **Step and Ramp Support**: Add steps and ramps to the pulse sequence.
+- **Compensation Pulse**: Keeps track of average voltage levels for compensation pulses.
+- **Configuration Management**: Updates the provided configuration with necessary operations and waveforms for the sequence.
+
+## Initialization
+
+To use the `VoltageGateSequence` class, start by initializing it with the necessary configuration and elements.
+
+```python
+from voltage_gate_sequence import VoltageGateSequence
+
+# Configuration dictionary for OPX
+configuration = {
+    "elements": {...},
+    "pulses": {...},
+    "waveforms": {...}
+}
+
+# List of elements involved in the gate operations
+elements = ["P5_sticky", "P6_sticky"]
+
+# Initialize the VoltageGateSequence
+seq = VoltageGateSequence(configuration, elements)
+```
+
+## Methods
+
+### `add_step`
+
+Add a voltage level to the pulse sequence. This function allows you to specify a voltage level and duration for each gate, with optional ramping to the desired level.
+
+```python
+def add_step(
+    self,
+    level: list[Union[int, QuaExpression, QuaVariable]] = None,
+    duration: Union[int, QuaExpression, QuaVariable] = None,
+    voltage_point_name: str = None,
+    ramp_duration: Union[int, QuaExpression, QuaVariable] = None
+) -> None:
+```
+
+- **level**: Desired voltage level of the different gates composing the gate in Volts.
+- **duration**: How long the voltage level should be maintained in ns. Must be a multiple of 4ns and larger than 16ns.
+- **voltage_point_name**: Name of the voltage level if added to the list of relevant points in the charge stability map.
+- **ramp_duration**: Duration in ns of the ramp if the voltage should be ramped to the desired level instead of stepped. Must be a multiple of 4ns and larger than 16ns.
+
+**Example Usage**:
+
+```python
+# Define points in the charge stability map [V1, V2]
+level_dephasing = [-0.07, 0.25]
+duration_dephasing = 2000  # nanoseconds
+
+# Add a dephasing step
+seq.add_step(level=level_dephasing, duration=duration_dephasing)
+
+# Add a step using a voltage point name and ramp duration
+seq.add_step(voltage_point_name="dephasing", ramp_duration=200)
+```
+
+In the example above:
+- The first call to `add_step` adds a dephasing step with a specified voltage level and duration.
+- The second call adds a step using a predefined voltage point named "dephasing" with a specified ramp duration.
+
+### `add_compensation_pulse`
+
+Add a compensation pulse of the specified duration whose amplitude is derived from the previous operations. This ensures that the average voltage seen by the gates remains zero over time.
+
+```python
+def add_compensation_pulse(self, duration: int) -> None:
+```
+
+- **duration**: Duration of the compensation pulse in clock cycles (4ns). Must be larger than 4 clock cycles.
+
+**Example Usage**:
+
+```python
+# Add a compensation pulse with a duration of 1000 ns
+seq.add_compensation_pulse(duration=1000)
```

Comment:
I think that this can be improved by setting the amplitude of the pulse instead and have the duration derived behind the scene (with slight adjustment of the amplitude for the duration to be a multiple of 4ns). Otherwise you can get an error because the needed amplitude is larger than 0.5V for the  specified duration...

Reply:
Are you proposing a change to the code or the README? I don't see how it's possible with the current API but I might be missing something.

Reply:
@TheoLaudatQM

Reply:
Oh sorry I missed this...
I meant in the code yes, I think it is just about modifying the function `add_compensation_pulse(self, duration: int)` so that the duration is inferred from the amplitude and not the opposite.
This being said it will require some time to test it properly so I suggest to add this functionality at a later stage in a new PR, what do you think?

Reply:
No problem. I think you're right that it's best if the duration is inferred from the amplitude, and not the opposite.

Let's leave it for future work so that we can merge this and integrate properly to qua-libs.

Reply:
@TheoLaudatQM could you please approve?